src/configuration/config.c

```diff
@@ -768,6 +863,30 @@ bool Config_Option_set(Config_Option_Field field, const char *val) {
 		}
 		break;
 
+		//----------------------------------------------------------------------
+		// timeout default
+		//----------------------------------------------------------------------
+
+		case Config_TIMEOUT_DEFAULT: {
+			long long timeout_default;
+			if(!_Config_ParseNonNegativeInteger(val, &timeout_default)) return false;
+			Config_timeout_default_set(timeout_default);
+			Config_log_if_old_and_new_timeout_used();
+		}
+		break;
```

Comment:
I think it would be wise to set `Config_TIMEOUT` to `CONFIG_TIMEOUT_NO_TIMEOUT` when setting `Config_TIMEOUT_DEFAULT` and vice versa, this will allow for easier retrieval of the timeout configuration and avoid conflicts, with this in-place the login in cmd_dispatch can be greatly simplified:
```c
Config_Option_get(Config_TIMEOUT, &a);
Config_Option_get(CONFIG_TIMEOUT_NO_TIMEOUT, &b);
write_timeout = b != Config_TIMEOUT_DEFAULT;
timeout = a | b;  // ok as at least one of them is 0
```